File: fleet-ui/packages/component/src/utils/case_format_utils.ts
import * as changeCase from "change-case";


// snake_case
// lower_underscore
export function lowerUnderscore(word: string): string {
  return changeCase.snakeCase(word);
}

// UpperCamelCase
// UpperCamel
export function upperCamel(word: string): string {
    return changeCase.pascalCase(word);
}

// UPPER_UNDERSCORE
export function UPPER_UNDERSCORE(word: string): string {
    return changeCase.constantCase(word);
}

// lowerCamelCase
// lowerCamel
export function lowerCamel(word: string): string {
    return changeCase.camelCase(word);
}

File: fleet-common/src/main/java/dev/osmanthus/fleet/common/utils/CaseFormatUtils.java
package dev.osmanthus.fleet.common.utils;

import com.google.common.base.CaseFormat;

public class CaseFormatUtils {

    // snake_case
    // lower_underscore
    public static String lowerUnderscore(String word) {
        return CaseFormat.LOWER_CAMEL.to(CaseFormat.LOWER_UNDERSCORE, word);
    }

    // UpperCamelCase
    // UpperCamel
    public static String upperCamel(String word) {
        return CaseFormat.LOWER_CAMEL.to(CaseFormat.UPPER_CAMEL, word);
    }

    // UPPER_UNDERSCORE
    public static String UPPER_UNDERSCORE(String word) {
        return CaseFormat.LOWER_CAMEL.to(CaseFormat.UPPER_UNDERSCORE, word);
    }

    // lowerCamelCase
    // lowerCamel
    public static String lowerCamel(String word) {
        return CaseFormat.UPPER_CAMEL.to(CaseFormat.LOWER_CAMEL, word);
    }


}
